### ISSUE TITLE
Multipart `$http.post()` in scripts

### ISSUE THREAD

**Issue by nihaals** [ProxymanApp/Proxyman#1906]
## Description

It would be great if there was a way to do a `multipart/form-data` request in scripts.

## Why this feature/change is important?

If you need to call an API that requires `multipart` you need to add `FormData` pollyfills which also might not interact with `$http.post()` correctly as the function itself isn't documented.

**Comment by NghiaTranUIT** [comment]
You're right @nihaals , the Scriprint tool doesn't support the FormData or `multipart`. The problem is JavascriptCore framework (from Apple) doesn't support axios, fetch, or URLSearchParam, or FormData.

I'm thinking if I can replace our custom `$http` with `whatwg-fetch` (https://www.npmjs.com/package/whatwg-fetch) to the Scripting, so it can work with FormData and other type of body.

**Comment by nihaals** [comment]
Being able to point to exactly what Proxyman uses sounds quite helpful as it would reduce the need for as thorough documentation on `$http` and especially if it's based on `fetch()`, its usage will be fairly intuitive and similar to JS people have already written. I also think trying to make Proxyman match a typical JS environment as much as possible makes it feel less like "it's JS syntax but a custom environment" and more like "it's just JS" (which documenting the environment scripts run in would also help with so users can dive into more docs if needed and search for workarounds).

**Comment by NghiaTranUIT** [comment]
@nihaals It's so difficult to support native `fetch` because `XMLHTTPRequest` doesn't support in JavascriptCore too.

Even with [React-native-fetch](https://github.com/react-native-community/fetch), they also develop their own custom `XMLHTTPRequest` with a lot of react-native logic. Tried to bring to Proxyman, but no luck 😢 

The same thing with axios, or superagent too, it's built on top of `XMLHTTPRequest`

**Comment by nihaals** [comment]
I guess you'd basically need to do something like https://github.com/Lukas-Stuehrk/XMLHTTPRequest? I have no idea how much work that is but I guess one advantage would be that if it passes tests you wouldn't have to mess with `$http` much again?

**Comment by NghiaTranUIT** [comment]
I tried, but it didn't work. The `fetch-polyfill` recognizes the XMLHttpRequest, but it stuck when making a request.

I guess this library is obsoleted (the last update was 7 years ago), and it only handles the Text-based Response body (https://github.com/Lukas-Stuehrk/XMLHTTPRequest/issues/1)

**Comment by nihaals** [comment]
Maybe a simpler solution would be adding `FormData` and supporting `$http.post({body: FormData})`?
